src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -211,9 +231,43 @@ public void configure(ActionMenuWidget actionMenu,
 			configureEditFileMetadataAction();
 			configureAddEvaluationAction();
 			configureCreateChallenge();
+			configureApproveUserAccess();
+		}
+	}
+	
+	
+	private void configureApproveUserAccess() {
+		actionMenu.setActionListener(Action.APPROVE_USER_ACCESS, this);
+		List<AccessRequirement> requirements = entityBundle.getAccessRequirements();
+		for (AccessRequirement ar : requirements) {
+			if (ar instanceof ACTAccessRequirement) {
+				actRequirements.add((ACTAccessRequirement) ar);
+			}
+		}
+		if (authenticationController.isLoggedIn()) {
+			
+			userProfileClient.getMyOwnUserBundle(IS_ACT_MEMBER, new AsyncCallback<UserBundle>() {
+				@Override
+				public void onSuccess(UserBundle userBundle) {
+					userBundle.setIsACTMember(true);
+					if (userBundle.getIsACTMember() && actRequirements.size() > 0) {
+						actionMenu.setActionVisible(Action.APPROVE_USER_ACCESS, true);
+						actionMenu.setActionEnabled(Action.APPROVE_USER_ACCESS, true);	
+					} else {
+						actionMenu.setActionVisible(Action.APPROVE_USER_ACCESS, false);

```

Comment:
by default, hide the action before rpc.  if act member (and there are act requirements) then show.